Dockerfile

```diff
@@ -1,13 +1,33 @@
-FROM node:18.17.0
+# Build
+FROM node:18.17.0-slim as build
 WORKDIR /app
 COPY . .
 
-ARG packageVersion=0.7.6
-ENV tarballFileName="gomboc-ai-cli-$packageVersion.tgz"
-ENV tarFileName="gomboc-ai-cli-$packageVersion.tar"
-
 RUN npm install
-RUN npm run build
-RUN npm pack
-RUN gzip -d $tarballFileName
-RUN npm install -g $tarFileName
\ No newline at end of file
+RUN npm run build 
+
+# Prod modules
+FROM node:18.17.0-slim as prod-modules
+WORKDIR /app
+COPY --from=build /app ./
+RUN npm ci --omit=dev
+
+RUN apt-get -y update \
+    && apt-get -y install git \
+    && npm link 
+
+# Production
+FROM gcr.io/distroless/nodejs18-debian12
+WORKDIR /app
+
+# Application copy
+COPY --from=prod-modules /usr/bin /usr/bin
```

Comment:
If you are going to blindly copy the distro then at this point distroless is likely getting in the way.

The `apt-get -y update` in the previous stage is what is getting rid of the CVEs, so my recommendation is to get rid of distroless entirely and have `prod-modules` be the final container.

Reply:
I would agree, however just using the image from prod-modules produces multiple high and critical CVEs so, even though its unclean, the final image produced by the distroless base images seems better.